### ISSUE TITLE
Some concerns about the implementation of the boundary loss

### ISSUE THREAD

**Issue by Rey-nard** [MedICL-VU/ProMISe#3]
Hello, I have some concerns about the implementation of the boundary loss.

According to the paper, boundary generator is applied to both ground truth and predicted mask, while the following code only applies it to the ground truth :
https://github.com/MedICL-VU/ProMISe/blob/2cf1b1e44e963d7cf833f198965d9130fd9ffebd/promise/src/processor/trainer.py#L101-L108

**Comment by HaoLi12345** [comment]
Hi, sorry for the late reply, and thanks for cathing this missing. I selected the wrong script to upload when creating this repo, and I have uploaded the one we used in our submission.
The "mask_loss" is used for approximate the binary mask since torch.argmax is non-differentiable.
After submission, I found directly use probabilities slightly improved the segmentation performance based on the mask_loss.
Let me know if there is any questions.

**Comment by HaoLi12345** [comment]
Hi, sorry for the late reply, and thanks for cathing this missing. I selected the wrong script to upload when creating this repo, and I have uploaded the one we used in our submission.
The "mask_loss" is used for approximate the binary mask since torch.argmax is non-differentiable.
After submission, I found directly use probabilities slightly improved the segmentation performance based on the mask_loss.
Let me know if there is any questions.
